src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+            patch_count_for_assessment, assessment_msg, under_internal_size_limit=False)
+
+        # Assert installation summary
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_SUCCESS,
+            patch_count_for_installation, installation_msg, substatus_index=1, under_internal_size_limit=False)
+
+        # Assert truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        assessment_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        installation_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data, substatus_index=1)
+        # Assert truncated status file size < completed status file size
+        self.assertTrue(len(json.dumps(truncated_substatus_file_data)) < len(json.dumps(complete_substatus_file_data)))
+
+        # Assert assessment truncation, keep min 5
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_ASSESSMENT_SUMMARY,
+            Constants.STATUS_WARNING, patch_count_for_assessment, assessment_truncated_msg)
+
+        # Assert all assessment fields in the message json are equal in both status files
+        self.__assert_assessment_truncated_msg_fields(assessment_msg, assessment_truncated_msg)
+
+        # Assert installation truncation
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_WARNING, patch_count_for_installation, installation_truncated_msg, substatus_index=1)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(installation_msg, installation_truncated_msg)
+
+    def test_exceptions_both_assessment_and_installation_truncated(self):
+        """ Perform truncation on assessment patches.
+        Input: xxx assessment patches
+        Output:
+        operation: Installation,
+        assessment substatus name: PatchAssessmentSummary,
+        complete status file size > 128kb,
+        truncated status file size < 126kb < 128kb,
+        truncated status file size < completed status file size,
+        assessment substatus status: warning,
+        assessment substatus patches < patch_count,
+        assessment errors code: 2 (warning),
+        assessment errors details count: 0,
+        assessment message json fields == truncated assessment message json fields.
+
+        Perform truncation on installation patches with multiple exceptions to ensure __try_add_error is working as expected and error code is 1 not 2 (warning)
```

Comment:
**__try_add_error** 

Is not in use anymore

Reply:
this unit test is needed because we need to assert when substatus status is error, we do not change it to warning, but i have combine this test with __assert_patch_summary_from_status and assert for truncation log.
done